File: app/src/main/java/com/botijasoftware/utils/IndexBufferByte.java
package com.botijasoftware.utils;


import java.nio.ByteBuffer;
import java.nio.ByteOrder;
import android.opengl.GLES20;

public class IndexBufferByte extends IndexBuffer {
	private ByteBuffer mBuffer;
	

	public IndexBufferByte(int type, int maxelements) {
		super(type, maxelements);
		if (maxelements > Byte.MAX_VALUE)
			this.maxelements = Byte.MAX_VALUE;
		format = GLES20.GL_BYTE;
		elementsize = Byte.SIZE/8;
		mBuffer = ByteBuffer.allocateDirect(maxelements*elementsize).order(
				ByteOrder.nativeOrder());

	}

	public ByteBuffer getBuffer() {
		return mBuffer;
	}
	
	public void put(int index, byte value) {
		mBuffer.put(index, value);
	}
	
	public void put(int index, short value) {
		mBuffer.put(index, (byte) value);
	}
	
	public void put(int index, int value) {
		mBuffer.put(index, (byte)value);
	}
		
	public void put(byte[] data) {
		mBuffer.put(data, 0, data.length);
	}
	
	public void put(byte value) {
		mBuffer.put(value);
	}
	
	public void put(short value) {
		mBuffer.put((byte) value);
	}
	
	public void put(int value) {
		mBuffer.put((byte) value);
	}
	
	public void position(int i) {
		mBuffer.position(i);
	}
	
	public int position() {
		return mBuffer.position();
	}

}

File: app/src/main/java/com/botijasoftware/utils/IndexBufferByte.kt
package com.botijasoftware.utils


import java.nio.ByteBuffer
import java.nio.ByteOrder
import android.opengl.GLES20

class IndexBufferByte(type: Int, maxelements: Int) : IndexBuffer(type, maxelements) {
    override val buffer: ByteBuffer


    init {
        if (maxelements > java.lang.Byte.MAX_VALUE)
            this.maxElements = java.lang.Byte.MAX_VALUE.toInt()
        format = GLES20.GL_BYTE
        elementsize = java.lang.Byte.SIZE / 8
        buffer = ByteBuffer.allocateDirect(maxelements * elementsize).order(
                ByteOrder.nativeOrder())

    }

    override fun put(index: Int, value: Byte) {
        buffer.put(index, value)
    }

    override fun put(index: Int, value: Short) {
        buffer.put(index, value.toByte())
    }

    override fun put(index: Int, value: Int) {
        buffer.put(index, value.toByte())
    }

    override fun put(data: ByteArray) {
        buffer.put(data, 0, data.size)
    }

    override fun put(value: Byte) {
        buffer.put(value)
    }

    override fun put(value: Short) {
        buffer.put(value.toByte())
    }

    override fun put(value: Int) {
        buffer.put(value.toByte())
    }

    override fun position(i: Int) {
        buffer.position(i)
    }

    override fun position(): Int {
        return buffer.position()
    }

}